pipeline/inputs/standard-input.md

```diff
@@ -1,18 +1,37 @@
 # Standard Input
 
-The **stdin** plugin allows to retrieve valid JSON text messages over the standard input interface \(stdin\). In order to use it, specify the plugin name as the input, e.g:
+The **stdin** plugin allows to retrieve a message stream from the standard input interface \(stdin\) of the fluent-bit process. In order to use it, specify the plugin name as the input, e.g:
 
 ```bash
 $ fluent-bit -i stdin -o stdout
 ```
 
-As input data the _stdin_ plugin recognize the following JSON data formats:
+If the stdin stream is closed (end-of-file), the stdin plugin will instruct
+`fluent-bit` to exit with success (0) after flushing any pending output.
+
+## Input formats
+
+If no parser is configured for the stdin plugin, it expects *valid json* input input data with one of the following data formats:
 
 ```bash
 1. { map => val, map => val, map => val }
 2. [ time, { map => val, map => val, map => val } ]
 ```
 
+It will log errors like
```

Comment:
I'm a little confused here - above it says to expect data in this format then it says there will be an error.

Can we clarify with correct format and then examples of failing ones?

Reply:
@patrick-stephens I can see how it would've been confusing.

* here's what you should provide
* here's the error you get if you violate that expectation

I'll rearrange.

Reply:
Rephrased to

---

If no parser is configured for the stdin plugin, it expects *valid json* input input data with one of the following data formats:
 
```bash
1. { map => val, map => val, map => val }
2. [ time, { map => val, map => val, map => val } ]                                           
```

Any input data that is *not* in one of the above formats will cause the plugin to log errors like: 
   
```
[error] [input:stdin:stdin.0] invalid record found, it's not a JSON map or array
```
 
To handle inputs in other formats, a parser must be explicitly specified in the configuration for the `stdin` plugin. See [parser input example](#parser-input-example) for sample configuration.